lib/rules/boolean-prop-naming.js

```diff
@@ -172,9 +172,10 @@ module.exports = {
      * @param {Function} addInvalidProp callback to run for each error
      */
     function runCheck(proptypes, addInvalidProp) {
-      proptypes = proptypes || [];
+      let newProptypes = proptypes;
+      newProptypes = newProptypes || [];
 
-      proptypes.forEach((prop) => {
+      newProptypes.forEach((prop) => {
```

Comment:
```suggestion
      (proptypes || []).forEach((prop) => {
```

Reply:
When I used this solution, an error arose saying that newProptypes is assigned a value but never used. Instead, I wrote it as (newProptypes || []).forEach((prop) => {

Reply:
Right - i don’t think we need a variable at all.

Reply:
@ljharb I have made the suggested changes if you wish to take a look at it.